lib/publiccloud/gcp_client.pm

```diff
@@ -71,36 +72,58 @@ sub get_next_vault_role {
 sub create_credentials_file {
     my ($self) = @_;
     my $credentials_file;
+    if (get_var('PUBLIC_CLOUD_CREDENTIALS_URL')) {
+        my $json = get_credentials();
+        # Need to remove quotes, convert \n to \\n and / to \/
+        assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\\"//g' | sed 's/\\\n/\\\\\\\n/g' | sed 's/\\//\\\\\//g')});
+        #assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\"//g' | sed 's/\\n/\\\\n/g'      | sed 's/\//\\\//g')});
 
-    if ($self->private_key_id()) {
+        $self->private_key_id(get_required_var('PUBLIC_CLOUD_PRIVATE_KEY_ID'));
         $credentials_file = "{" . $/
           . '"type": "service_account", ' . $/
           . '"project_id": "' . $self->project_id . '", ' . $/
           . '"private_key_id": "' . $self->private_key_id . '", ' . $/
-          . '"private_key": "' . $self->private_key . '", ' . $/
-          . '"client_email": "' . $self->service_acount_name . '@' . $self->project_id . '.iam.gserviceaccount.com", ' . $/
-          . '"client_id": "' . $self->client_id . '", ' . $/
-          . '"auth_uri": "https://accounts.google.com/o/oauth2/auth", ' . $/
-          . '"token_uri": "https://oauth2.googleapis.com/token", ' . $/
-          . '"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs", ' . $/
-          . '"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/' . $self->service_acount_name . '%40' . $self->project_id . '.iam.gserviceaccount.com"' . $/
+          . '"private_key": "PRIVATE_KEY", ' . $/
+          . '"client_email": "' . $self->service_acount_name . '@developer.gserviceaccount.com", ' . $/
+          . '"client_id": "' . $self->client_id . '" ' . $/
           . '}';
-    } else {
-        record_info('INFO', 'Get credentials from VAULT server.');
-
-        my $data = $self->vault->retry(
-            sub { $self->vault->get_secrets('/gcp/key/' . $self->get_next_vault_role(), max_tries => 1) },
-            name => 'get_secrets(gcp)',
-            max_tries => scalar($self->vault_gcp_roles()) * 2,
-            sleep_duration => get_var('PUBLIC_CLOUD_VAULT_TIMEOUT', 5));
-        $credentials_file = b64_decode($data->{private_key_data});
-        my $cf_json = decode_json($credentials_file);
-        $self->account($cf_json->{client_email});
-        $self->project_id($cf_json->{'project_id'});
+
+        save_tmp_file(CREDENTIALS_FILE, $credentials_file);
+        assert_script_run('curl ' . autoinst_url . '/files/' . CREDENTIALS_FILE . ' -o ' . CREDENTIALS_FILE);
+        script_output('sed -i "s/PRIVATE_KEY/$key/" ' . CREDENTIALS_FILE);
     }
 
-    save_tmp_file(CREDENTIALS_FILE, $credentials_file);
-    assert_script_run('curl ' . autoinst_url . '/files/' . CREDENTIALS_FILE . ' -o ' . CREDENTIALS_FILE);
+    else {
```

Comment:
no need to review this, this is just diff of moving the "old" block in the else.